File: gamepack/Decompiled/PacketWriter.java
import java.awt.Image;
import java.io.IOException;

public class PacketWriter {
   static Image pauseImage;
   boolean at = true;
   int aa = 0;
   int al = 0;
   int serverPacket0Length = 0;
   int ay = 0;
   ServerProt ai;
   ServerProt ao;
   ServerProt serverPacket0 = null;
   ServerProt ax;
   IterableNodeDeque bitNodes = new IterableNodeDeque();
   AbstractSocket socket0;
   Packet packet = new Packet(5000);
   PacketBit bit = new PacketBit(40000);
   public Isaac au;

   PacketWriter() {
   }

   final void af() {
      this.bitNodes.clear0();
      this.serverPacket0Length = 0;
   }

   final void an() throws IOException {
      if (null != this.socket0 && -1859952183 * this.serverPacket0Length > 0) {
         this.packet.index = 0;

         while(true) {
            PacketBitNode var2 = (PacketBitNode)this.bitNodes.last();
            if (null == var2 || 1816684323 * var2.ac > this.packet.array.length - this.packet.index * -1633313603) {
               this.socket0.ab(this.packet.array, 0, this.packet.index * -1633313603);
               this.ay = 0;
               break;
            }

            this.packet.cs(var2.bit.array, 0, 1816684323 * var2.ac);
            this.serverPacket0Length -= -56666229 * var2.ac;
            var2.remove();
            var2.bit.bg();
            var2.clearAll();
         }
      }

   }

   public final void aw(PacketBitNode var1) {
      this.bitNodes.addFirst(var1);
      var1.ac = -1120134497 * var1.bit.index;
      var1.bit.index = 0;
      this.serverPacket0Length += var1.ac * -56666229;
   }

   void setSocket(AbstractSocket var1) {
      this.socket0 = var1;
   }

   void au() {
      if (this.socket0 != null) {
         this.socket0.aq();
         this.socket0 = null;
      }

   }

   void ab() {
      this.socket0 = null;
   }

   AbstractSocket getSocket() {
      return this.socket0;
   }
}

File: find-names/src/main/java/mapper/identifiers/classes/PacketWriter.kt
package mapper.identifiers.classes

import mapper.abstractclasses.IdentityMapper
import mapper.abstractclasses.OrderMapper
import mapper.abstractclasses.UniqueMapper
import mapper.annotations.DependsOn
import mapper.annotations.MethodParameters
import mapper.identifiers.classes.*
import mapper.predicateutilities.and
import mapper.predicateutilities.predicateOf
import mapper.predicateutilities.type
import mapper.wrappers.ClassWrapper
import mapper.wrappers.FieldWrapper
import mapper.wrappers.InstructionWrapper
import mapper.wrappers.MethodWrapper
import org.objectweb.asm.Opcodes
import org.objectweb.asm.Type
import org.runestar.client.common.startsWith

@DependsOn(AbstractSocket::class)
class PacketWriter : IdentityMapper.Class() {

    override val predicate = predicateOf<ClassWrapper> { it.superType == Any::class.type }
            .and { it.instanceFields.any { it.type == type<AbstractSocket>() } }

    @DependsOn(AbstractSocket::class)
    class socket0 : IdentityMapper.InstanceField() {
        override val predicate = predicateOf<FieldWrapper> { it.type == type<AbstractSocket>() }
    }

    @MethodParameters()
    @DependsOn(AbstractSocket::class)
    class getSocket : IdentityMapper.InstanceMethod() {
        override val predicate = predicateOf<MethodWrapper> { it.returnType == type<AbstractSocket>() }
    }

    @MethodParameters("socket")
    @DependsOn(AbstractSocket::class)
    class setSocket : IdentityMapper.InstanceMethod() {
        override val predicate = predicateOf<MethodWrapper> { it.returnType == Type.VOID_TYPE }
                .and { it.arguments.startsWith(type<AbstractSocket>()) }
    }


    @DependsOn(Packet::class)
    class packet : IdentityMapper.InstanceField() {
        override val predicate = predicateOf<FieldWrapper> { it.type == type<Packet>() }
    }

    @DependsOn(PacketBit::class)
    class bit : IdentityMapper.InstanceField() {
        override val predicate = predicateOf<FieldWrapper> { it.type == type<PacketBit>() }
    }

    @DependsOn(IterableNodeDeque::class)
    class bitNodes : IdentityMapper.InstanceField() {
        override val predicate = predicateOf<FieldWrapper> { it.type == type<IterableNodeDeque>() }
    }


//    @DependsOn(socket0::class)
//    @MethodParameters()
//    class removeSocket : IdentityMapper.InstanceMethod() {
//        override val predicate = predicateOf<Method2> { it.returnType == Type.VOID_TYPE }
//                .and { it.arguments.size in 0..1 }
//                .and { it.instructions.any { it.opcode == Opcodes.PUTFIELD && it.fieldId == field<socket0>().id } }
//                .and { it.instructions.any { it.opcode == Opcodes.ACONST_NULL } }
//                .and { it.instructions.none { it.isMethod } }
//    }
//    @DependsOn(Isaac::class)
//    class isaac : IdentityMapper.InstanceField() {
//        override val predicate = predicateOf<Field2> { it.type == type<Isaac>() }
//    }

//    @MethodParameters()
//    @DependsOn(AbstractSocket.close::class)
//    class close : IdentityMapper.InstanceMethod() {
//        override val predicate = predicateOf<Method2> { it.returnType == VOID_TYPE }
//                .and { it.instructions.any { it.isMethod && it.methodId == method<AbstractSocket.close>().id } }
//    }

    @DependsOn(ServerProt::class)
    class serverPacket0 : UniqueMapper.InConstructor.Field(PacketWriter::class) {
        override val predicate = predicateOf<InstructionWrapper> { it.opcode == Opcodes.PUTFIELD && it.fieldType == type<ServerProt>() }
    }

    class serverPacket0Length : OrderMapper.InConstructor.Field(PacketWriter::class, 1) {
        override val predicate = predicateOf<InstructionWrapper> { it.opcode == Opcodes.PUTFIELD && it.fieldType == Type.INT_TYPE }
    }
}